fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/QueryResult.java

```diff
@@ -44,28 +54,31 @@ public class QueryResult {
     @Nullable
     private final Object datum;
 
+    // TODO: this can be removed because it is (probably) redundant. The `queriedRecord` is mostly kept here to flow
+    //       information that already exist in the `datum` such as the primary key constituents. There are however
+    //       some extra information that are exclusive to the `queriedRecord` such as the record type and the record
+    //       version, and the size, however this information can be carried out through modelling the plan operators
+    //       differently such that they include any relevant information (such as record version) in the flown result
+    //       (i.e. the `datum`) itself.
     @Nullable
     private final FDBQueriedRecord<?> queriedRecord;
 
-    @Nullable
-    private final IndexEntry indexEntry;
-
```

Comment:
After analyzing the usages of this field I went ahead and removed it because it was either set to null, or set to the _same_ `IndexEntry` field of the `queriedRecord`.